readium/streamer/src/main/java/org/readium/r2/streamer/fetcher/HtmlInjector.kt

```diff
@@ -42,7 +42,7 @@ internal class HtmlInjector(
     private suspend fun inject(resource: Resource): Resource = object : TransformingResource(resource) {
 
         override suspend fun transform(data: ResourceTry<ByteArray>): ResourceTry<ByteArray> =
-            resource.read().mapCatching {
```

Comment:
Ups, we were reading the full resource instead of transforming the given data, which read the resource twice.